cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt

```diff
@@ -102,27 +105,60 @@ private class NodeRunner(baseDebugPort: Int, baseMonitoringPort: Int) {
     }
 
     private fun startWindowed(command: List<String>, workingDir: File, nodeName: String): Process {
+
+        val createTempScriptFile = {
+            val tmpFile: File = File.createTempFile("corda_node_script_launcher", "")
+            BufferedWriter(OutputStreamWriter(FileOutputStream(tmpFile))).use {
+                //Start shell to keep window open unless java terminated normally or due to SIGTERM:
+                it.write("""
+                        #!/bin/sh
+                        set -v
+                        unlink $tmpFile
```

Comment:
Why does this script want to delete itself?

Reply:
because it a temporary file that needs to be deleted by someone, it cannot be deleted by the Java process using `File#deleteOnExit` because there would be a race condition with the Java process terminating before the shell process is actually fired up (it's a fork so there is no synchronization guarantee)

Reply:
I think it only "needs" deleting because you keep generating new ones with different values hard-coded into them. Have you tried just creating _one_ script, and passing the values in as parameters?

But this PR has bigger problems than that anyway: The `runnodes` script no longer launches nodes on my machine. Instead, it seems just to dump all of the different `$shellScript` contents to the console and then exit.